query-node/mappings/src/storage.ts

```diff
@@ -161,6 +165,46 @@ async function updateSingleConnectedAsset<T extends Channel | Video>(db: Databas
   }
 }
 
+// removes connection between dataObject and other entities
+async function disconnectDataObjectRelations(db: DatabaseManager, dataObject: DataObject) {
+  await disconnectSingleDataObjectRelation(db, new Channel(), 'avatarPhoto', dataObject)
+  await disconnectSingleDataObjectRelation(db, new Channel(), 'coverPhoto', dataObject)
+
+  await disconnectSingleDataObjectRelation(db, new Video(), 'thumbnailPhoto', dataObject)
+  await disconnectSingleDataObjectRelation(db, new Video(), 'media', dataObject)
+}
+
+async function disconnectSingleDataObjectRelation<T extends Channel | Video>(db: DatabaseManager, type: T, propertyName: string, dataObject: DataObject) {
+  // prepare lookup condition
+  const condition = {
+    where: {
+      [propertyName + 'DataObject']: dataObject
+    }
+  } // as FindConditions<T>
+
+  // in therory the following condition(s) can be generalized `... db.get(type, ...` but in practice it doesn't work :-\
+  const items = type instanceof Channel
+    ? await db.getMany(Channel, condition)
+    : await db.getMany(Video, condition)
+
+  for (const item of items) {
+    item[propertyName + 'Availability'] = AssetAvailability.INVALID
```

Comment:
How does Atlas handle INVALID case?
Should we probably add a new variant `AssetAvailability.Deleted` ?

cc: @kdembler

Reply:
cc: @bwhm

Reply:
It seems that `Deleted` flag is currently not needed by Atlas.

Reply:
Just to reiterate, Atlas treats `Invalid` as "not provided" or "missing". I don't think we need a separate status for deleted